modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/service/AsahSegmentsEntryLocalServiceWrapper.java

```diff
@@ -105,11 +105,38 @@ public SegmentsEntry recalculateSegmentsEntry(long segmentsEntryId)
 		Criteria criteria = expressionContext.accept(
 			new IndividualSegmentsExpressionVisitorImpl());
 
-		return updateSegmentsEntry(
+		return super.updateSegmentsEntry(
 			segmentsEntry.getSegmentsEntryId(), individualSegment.getId(),
 			nameMap, null, true, _serialize(criteria), serviceContext);
 	}
 
+	@Override
+	public SegmentsEntry updateSegmentsEntry(
+			long segmentsEntryId, String segmentsEntryKey,
+			Map<Locale, String> nameMap, Map<Locale, String> descriptionMap,
+			boolean active, String criteria, ServiceContext serviceContext)
+		throws PortalException {
+
```

Comment:
We should check if the FF is enabled and if analytics is enabled also.

Reply:
As far as I could check, I agree with Mikel, this method is only called from the SegmentsEntryLocalServiceWrapper and there, it is called from a bunch of different places. Just to be sure and avoid possible misbehabiour in the future we should do both checks